### PR TITLE
[NO SQUASH] FPS limiter / Windows fixes

### LINKED ISSUE #11891: FPS is consistently below fps_max even when drawtime is well below the limit
##### Minetest version

```
Minetest 5.4.1 (Windows)
Using Irrlicht 1.8.4
Using LuaJIT 2.1.0-beta3
BUILD_TYPE=Release
RUN_IN_PLACE=1
USE_CURL=1
USE_GETTEXT=1
USE_SOUND=1
USE_FREETYPE=1
STATIC_SHAREDIR="."
STATIC_LOCALEDIR="locale"
```

##### OS / Hardware

Operating system: Windows 10
CPU: Intel Core i7-9700 Pro

GPU model: NVIDIA GeForce RTX 2070
OpenGL version: OpenGL 4.6.0

##### Summary

The fps_max setting is inaccurate, and the actual limit seems to depend on how long it takes to draw a frame. A drawtime close to the limit allows rendering FPS to reach values near (but not quite) fps_max. A low drawtime causes FPS to be consistently limited to a value much lower than fps_max (~40 for fps_max = 60, and ~64 for fps_max = 90).

Turning off vertical sync through the Minetest setting does not affect the bug. Turning off vsync in the NVIDIA Control Panel does not have any effect either.

Bug occurs on several Minetest versions on Windows 10.
Bug does not seem to occur in Minetest 5.5.0-dev or 5.4.2 on Ubuntu 21.10 (using OpenGL 4.3).

##### Steps to reproduce

This happens on all singleplayer worlds I tried, and also when hosting on one window and joining 127.0.0.1 in a second window. Joining a server had the same effect. Here is the best way to demonstrate the problem:
1. Join any performance intensive world (lots of mods, nodes with complex meshes, etc.).
2. Walk into a wall so that Minetest renders as little as possible. Drawtime plummets, but FPS is much lower than fps_max.
3. Slowly back away from the wall so that, gradually, more of the performance intensive world is rendered.
4. FPS will increase up until your computer is only just able to render at fps_max. Past that point, the effect will disappear and FPS will decrease as usual.

### LINKED ISSUE #11981: windows performance enhancements
On Windows, Minetest sometimes runs slowly due to Windows deciding to use the integrated GPU rather than the discrete GPU. I found an article about optimization methods for windows, one of which is to hint to the driver to use the discrete GPU by exporting a particular 4-byte variable.

https://www.anthropicstudios.com/2022/01/13/asking-windows-nicely/

See also: #10576

### PR DESCRIPTION
**Pull request by sfan5**
should fix #11891

fixes #11981

## To do

This PR is Ready for Review.

### CODE REVIEW

**Comment by sfan5** [comment]
Tested on an old Intel+NVIDIA Windows 7 laptop. The behavior described in #11891 is not reproducible.

**Comment by kilbith** [comment]
> Tested on an old Intel+NVIDIA Windows 7 laptop. The behavior described in #11891 is not reproducible.

The issue says Windows 10 and up. Not Windows 7 (EOL OS).

**Comment by sfan5** [comment]
I know, but (if I've diagnosed it correctly) the issue and fix both apply to older versions too. And as always community members can help by testing on their own setup.

File: src/client/game.cpp [hunk 1]
```diff
@@ -4045,47 +4053,46 @@ void Game::updateShadows()
...
+	last_time = porting::getTimeUs();
+}
 
-	u32 frametime_min = 1000 / (
+/*
+ * On some computers framerate doesn't seem to be automatically limited
+ */
+void FpsControl::limit(IrrlichtDevice *device, f32 *dtime)
+{
+	const u64 frametime_min = 1000000.0f / (
```
Comment (SmallJoker) [inline src/client/game.cpp:4066]: Idea: `(float)1E6` would ensure a correct quantity of zeros. Same for the change below.

**Review by SmallJoker — APPROVED** [review]
Works on Linux.

**Pull request merged**